homeassistant/components/switch/template.py

```diff
@@ -25,40 +27,29 @@
 _LOGGER = logging.getLogger(__name__)
 _VALID_STATES = [STATE_ON, STATE_OFF, 'true', 'false']
 
+SWITCH_SCHEMA = vol.Schema({
+    vol.Required(CONF_VALUE_TEMPLATE): cv.template,
+    vol.Required(ON_ACTION): cv.SCRIPT_SCHEMA,
+    vol.Required(OFF_ACTION): cv.SCRIPT_SCHEMA,
+    vol.Optional(ATTR_FRIENDLY_NAME): cv.string,
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SWITCHES): vol.Schema({cv.slug: SWITCH_SCHEMA}),
+})
+
 
 # pylint: disable=unused-argument
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup the Template switch."""
     switches = []
-    if config.get(CONF_SWITCHES) is None:
-        _LOGGER.error("Missing configuration data for switch platform")
-        return False
 
     for device, device_config in config[CONF_SWITCHES].items():
-
-        if device != slugify(device):
-            _LOGGER.error("Found invalid key for switch.template: %s. "
-                          "Use %s instead", device, slugify(device))
-            continue
-
-        if not isinstance(device_config, dict):
-            _LOGGER.error("Missing configuration data for switch %s", device)
-            continue
-
         friendly_name = device_config.get(ATTR_FRIENDLY_NAME, device)
         state_template = device_config.get(CONF_VALUE_TEMPLATE)
```

Comment:
It's not critical, but you can use `device_config[CONF_VALUE_TEMPLATE]` when you know the value will be there